### PR TITLE
Fixes #19178: Add rudder http client to rudder agent

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/19178

### CODE REVIEW

**Comment by amousset** [comment]
We also need to make the port configurable

**Comment by peckpeck** [comment]
PR updated with a new commit

File: bin/rudder-client [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+fi
+if [ "${PINNED_HASH}" != "" ]; then
+  SECOPTS="--pinnedpubkey ${PINNED_HASH} --insecure"
+elif [ "${VERIFY_CERTIFICATES}" = "true" ]; then
+  SECOPTS=""
+else
+  SECOPTS="--insecure"
+fi
+
+# get server authentication from russer.json if available and not overridden
```
Comment (amousset) [inline bin/rudder-client:70]:
```suggestion
# get server authentication from rudder.json if available and not overridden
```

**Review by amousset — APPROVED** [review]

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Pull request merged**